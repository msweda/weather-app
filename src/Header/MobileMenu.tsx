import { RootRoute } from '@/common/enums'
import { browserHistory } from '@/common/utils'
import Drawer from '@material-ui/core/Drawer'
import IconButton from '@material-ui/core/IconButton'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import {
  createStyles,
  Theme,
  WithStyles,
  withStyles,
} from '@material-ui/core/styles'
import MenuIcon from '@material-ui/icons/Menu'
import * as React from 'react'
import mobileMenuItems from './mobileMenuItems'

class MobileMenu extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      isOpen: false,
    }
  }

  render() {
    const { classes } = this.props
    return (
      <div>
        <IconButton onClick={this.handleOpen}>
          <MenuIcon className={classes.menuButton} />
        </IconButton>
        <Drawer
          anchor="right"
          open={this.state.isOpen}
          onClose={this.handleClose}
        >
          <List>
            {mobileMenuItems.map((item, index) => (
              <ListItem
                key={`mobileMenu-listItem-${index}`}
                button={true}
                onClick={this.handleMenuItemClick.bind(this, item.route)}
              >
                <ListItemIcon>{item.icon}</ListItemIcon>
                <ListItemText secondary={item.label} />
              </ListItem>
            ))}
          </List>
        </Drawer>
      </div>
    )
  }

  handleOpen = () => {
    this.setState({ isOpen: true })
  }

  handleClose = () => {
    this.setState({ isOpen: false })
  }

  handleMenuItemClick = (route: RootRoute) => {
    browserHistory.push(route)
    this.handleClose()
  }
}

const styles = (theme: Theme) =>
  createStyles({
    menuButton: {
      color: theme.palette.background.paper,
    },
  })

interface Props extends WithStyles<typeof styles> {}
interface State {
  isOpen: boolean
}

export default withStyles(styles)(MobileMenu)
