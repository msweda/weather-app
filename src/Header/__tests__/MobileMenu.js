import Drawer from '@material-ui/core/Drawer'
import IconButton from '@material-ui/core/IconButton'
import List from '@material-ui/core/List'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import { createShallow } from '@material-ui/core/test-utils'
import AcUnitIcon from '@material-ui/icons/AcUnit'
import Filter5Icon from '@material-ui/icons/Filter5'
import MenuIcon from '@material-ui/icons/Menu'
import * as React from 'react'
import MobileMenu from '../MobileMenu'

describe('<MobileMenu />', () => {
  let shallow
  let wrapper

  beforeEach(() => {
    shallow = createShallow({ untilSelector: MobileMenu })
    wrapper = shallow(<MobileMenu />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render <MenuIcon /> inside an <IconButton />', () => {
    const iconButton = wrapper.find(IconButton)
    expect(iconButton.exists()).toBeTruthy()
    expect(
      iconButton
        .children()
        .find(MenuIcon)
        .exists()
    ).toBeTruthy()
  })

  it('should render a <List /> of mobile menu items inside a <Drawer />', () => {
    const drawer = wrapper.find(Drawer)
    expect(drawer.exists()).toBeTruthy()
    expect(
      drawer
        .children()
        .find(List)
        .exists()
    ).toBeTruthy()
  })

  it('should render a <ListItemIcon /> and a <ListItemText /> for the weather page', () => {
    const listItemIcon = wrapper.find(ListItemIcon).at(0)
    expect(listItemIcon.exists()).toBeTruthy()
    const icon = listItemIcon.children().find(AcUnitIcon)
    expect(icon.exists()).toBeTruthy()
    const text = wrapper
      .find(ListItemText)
      .at(0)
      .props().secondary
    expect(text).toEqual('Weather')
  })

  it('should render a <ListItemIcon /> and a <ListItemText /> for the forecast page', () => {
    const listItemIcon = wrapper.find(ListItemIcon).at(1)
    expect(listItemIcon.exists()).toBeTruthy()
    const icon = listItemIcon.children().find(Filter5Icon)
    expect(icon.exists()).toBeTruthy()
    const text = wrapper
      .find(ListItemText)
      .at(1)
      .props().secondary
    expect(text).toEqual('Forecast')
  })
})
