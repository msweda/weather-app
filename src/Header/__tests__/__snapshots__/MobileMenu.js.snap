// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileMenu /> should match snapshot 1`] = `
<div>
  <WithStyles(IconButton)
    onClick={[Function]}
  >
    <pure(MenuIcon)
      className="MobileMenu-menuButton-1"
    />
  </WithStyles(IconButton)>
  <WithStyles(Drawer)
    anchor="right"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(List)>
      <WithStyles(ListItem)
        button={true}
        key="mobileMenu-listItem-0"
        onClick={[Function]}
      >
        <WithStyles(ListItemIcon)>
          <pure(AcUnitIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          secondary="Weather"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        button={true}
        key="mobileMenu-listItem-1"
        onClick={[Function]}
      >
        <WithStyles(ListItemIcon)>
          <pure(Filter5Icon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          secondary="Forecast"
        />
      </WithStyles(ListItem)>
    </WithStyles(List)>
  </WithStyles(Drawer)>
</div>
`;
